integration-tests/pack-tests/src/test/java/org/apache/servicecomb/saga/integration/pack/tests/PackIT.java

```diff
@@ -72,90 +69,92 @@ public void tearDown() throws Exception {
 
   @Test
   public void updatesTxStateToAlpha() throws Exception {
-    HttpHeaders headers = new HttpHeaders();
-
-    headers.set(OmegaContext.GLOBAL_TX_ID_KEY, globalTxId);
-
-    ResponseEntity<String> entity = restTemplate.exchange("/greet?name={name}",
-        GET,
-        new HttpEntity<>(headers),
+    ResponseEntity<String> entity = restTemplate.getForEntity("/greet?name={name}",
         String.class,
         "mike");
 
     assertThat(entity.getStatusCode(), is(OK));
     assertThat(entity.getBody(), is("Greetings, mike; Bonjour, mike"));
 
+    List<String> distinctGlobalTxIds = repository.findDistinctGlobalTxId();
+    assertThat(distinctGlobalTxIds.size(), is(1));
+
+    String globalTxId = distinctGlobalTxIds.get(0);
     List<TxEventEnvelope> envelopes = repository.findByGlobalTxIdOrderByCreationTime(globalTxId);
 
-    assertThat(envelopes.size(), is(4));
-    assertThat(envelopes.get(0).type(), is("TxStartedEvent"));
+    assertThat(envelopes.size(), is(5));
+    assertThat(envelopes.get(0).type(), is("SagaStartedEvent"));
```

Comment:
please rename these `envelopes.get(i)` to sagaStartedEvent, txStartedEvent, etc. it's getting harder to read.